Formula/g/guichan.rb

```diff
@@ -0,0 +1,168 @@
+class Guichan < Formula
+  desc "Small, efficient C++ GUI library designed for games"
+  homepage "https://github.com/darkbitsorg/guichan"
+  url "https://github.com/darkbitsorg/guichan/releases/download/v0.8.3/guichan-0.8.3.tar.gz"
+  sha256 "2f3b265d1b243e30af9d87e918c71da6c67947978dcaa82a93cb838dbf93529b"
+  license "BSD-3-Clause"
+
+  bottle do
+    sha256 cellar: :any,                 arm64_sonoma:   "f9946ca2ce53833770ba41ce2f9fc5b630e46d595b1aaf893e8be3a302841dda"
+    sha256 cellar: :any,                 arm64_ventura:  "ff8971a12e820a77df5f570dbe145fb5f1046d851d713d8f26f89030429ea93a"
+    sha256 cellar: :any,                 arm64_monterey: "917384f24bef699687d0cdd48867f03733db6aeee593537da519360e250fd22e"
+    sha256 cellar: :any,                 arm64_big_sur:  "3acc3607e1930e9864934244b269370ffc35081c10138daf9b61254195bacc7f"
+    sha256 cellar: :any,                 sonoma:         "e5a2bcb4611579d22df6474a8986d6127478a4ab9b14dc4bca10cbeb490bd9b1"
+    sha256 cellar: :any,                 ventura:        "5415af4555a0b2bbacc69dfba87485659a55dc3d7a80c24599a172036235da5d"
+    sha256 cellar: :any,                 monterey:       "1ef2ef362f796f72ba510c5e1878e8b290846bec405e7b5240e8485971ae6950"
+    sha256 cellar: :any,                 big_sur:        "2d4f9b296640bffe66b4eb09642ab499517d050821de8299da838937f8611542"
+    sha256 cellar: :any,                 catalina:       "93a5e8526479a48a82a7890393b2e8871e3cb2e4dae4fae4bddb964177fe784e"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:   "654afd1dcf4f1fa523bba39ded7799d50ec10918eeece0aa5274ba123de17cea"
+  end
+
+  on_linux do
+    depends_on "mesa"
+    depends_on "mesa-glu"
+  end
+
+  def install
+    system "./configure", *std_configure_args
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"helloworld.cpp").write <<~EOS
```

Comment:
Test file is way too long. 
If this is from the code repository, you can install them from `install` block with
```
  pkgshare.install "examples/some_files"
```
and then load it from `test` block.

Here is an example.
- https://github.com/Homebrew/homebrew-core/blob/c457554a0bac06bbc47e8de68065f814a87c17ac/Formula/b/bcftools.rb#L1

Reply:
I wrote a new test because the previous one depended on SDL1. I plan to add it to the Guichan repository, so I can use it from there. Thanks!

Reply:
I'll keep this in mind for the next release, but since no such example is included in the current release I've opted to reduce the test code to be rather more minimal. It's only a test anyway and not meant as an example of how to use Guichan.